render/image.go

```diff
@@ -1,37 +1,61 @@
 package render
 
 import (
+	"errors"
 	"fmt"
 	"image"
 	"image/color"
 	"image/draw"
+	"io/fs"
+	"io/ioutil"
 	"os"
+	"path/filepath"
 	"strings"
-	"sync"
+	"time"
 
+	"github.com/BattlesnakeOfficial/exporter/inkscape"
+	"github.com/BattlesnakeOfficial/exporter/media"
 	"github.com/disintegration/imaging"
 	"github.com/fogleman/gg"
-	log "github.com/sirupsen/logrus"
+	"github.com/patrickmn/go-cache"
+	"github.com/sirupsen/logrus"
 )
 
+// AssetType is intended to be used like an enum for the type of asset we are loading
+type AssetType string
+
+const (
+	// AssetHead is a snake head
+	AssetHead AssetType = "head"
+	// AssetTail is a snake tail
+	AssetTail AssetType = "tail"
+)
+
+type rotations int
+
 const (
-	AssetFallbackHead        = "heads/regular.png"
-	AssetFallbackTail        = "tails/regular.png"
-	AssetFallbackUnspecified = ""
-	BoardBorder              = 2
-	SquareSizePixels         = 20
-	SquareBorderPixels       = 1
-	SquareFoodRadius         = SquareSizePixels / 3
-	ColorEmptySquare         = "#f0f0f0"
-	ColorFood                = "#ff5c75"
-	ColorHazard              = "#00000066"
+	rotate0 rotations = iota
+	rotate90
+	rotate180
+	rotate270
 )
 
-var boardImageCache = make(map[string]image.Image)
-var boardImageCacheLock sync.Mutex
+const (
+	AssetFallbackHeadName = "default"
+	AssetFallbackTailName = "default"
+	BoardBorder           = 2
+	SquareSizePixels      = 20
+	SquareBorderPixels    = 1
+	SquareFoodRadius      = SquareSizePixels / 3
+	ColorEmptySquare      = "#f0f0f0"
+	ColorFood             = "#ff5c75"
+	ColorHazard           = "#00000066"
+)
 
-var assetImageCache = make(map[string]image.Image)
-var assetImageCacheLock sync.Mutex
+var boardImageCache = cache.New(6*time.Hour, time.Minute)
```

Comment:
use the cache library instead of managing our own implementation of a cache

Reply:
See note about possibly burying this inside the media client